### PR TITLE
Dsylaiev/tdi 42576 refactor azure common

### PR DESCRIPTION
**Pull request by dmytro-sylaiev**

### CODE REVIEW

**Comment by undx** [comment]
Hello @dmytro-sylaiev, can you rebase your branch?

File: azure/azurecommon/src/main/resources/org/talend/components/azure/common/Messages.properties [hunk 1]
```diff
@@ -1,2 +0,0 @@
-Protocol.HTTP._displayName=HTTP
-Protocol.HTTPS._displayName=HTTPS
```
Comment (undx) [inline azure/azurecommon/src/main/resources/org/talend/components/azure/common/Messages.properties]: These resources are missing and validation cannot pass.
```azurecommon has validation errors:
Missing key Protocol.HTTP._displayName in class org.talend.components.common.connection.azurestorage.Protocol resource bundle
Missing key Protocol.HTTPS._displayName in class org.talend.components.common.connection.azurestorage.Protocol resource bundle
Missing key org.talend.components.common.service.azurestorage.MessageService.invalidAccountKeyFormat in interface org.talend.components.common.service.azurestorage.MessageService resource bundle``

File: azure/pom.xml [hunk 2]
```diff
@@ -46,6 +42,76 @@
             <version>8.1.0</version>
```
Comment (undx) [inline azure/pom.xml]: in properties?

File: azure/pom.xml [hunk 3]
```diff
@@ -46,6 +42,76 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-lang3</artifactId>
+            <version>${common.lang.version}</version>
+        </dependency>
+
         <dependency>
             <groupId>com.fasterxml.jackson.core</groupId>
             <artifactId>jackson-core</artifactId>
```
Comment (undx) [inline azure/pom.xml]: Plugins below are already defined in parent pom (version,parms, etc.), is it needed to fully re-insert them?

File: azure/azure-dls-gen2/pom.xml [hunk 4]
```diff
@@ -14,14 +14,15 @@
...
-        <avro.version>1.8.2</avro.version>
-        <hadoop-common.version>2.8.2</hadoop-common.version>
         <hamcrest.version>1.3</hamcrest.version>
     </properties>
 
     <dependencies>
+        <dependency>
+            <groupId>org.talend.components</groupId>
+            <artifactId>azurecommon</artifactId>
+            <version>1.3.0-SNAPSHOT</version>
```
Comment (undx) [inline azure/azure-dls-gen2/pom.xml]: Maybe set a property for the version defined in parent module (azure).

File: azure/azureblob/pom.xml [hunk 5]
```diff
@@ -52,71 +30,6 @@
             <artifactId>azurecommon</artifactId>
             <version>1.3.0-SNAPSHOT</version>
```
Comment (undx) [inline azure/azureblob/pom.xml]: same remark.

File: azure/azureblob/pom.xml [hunk 6]
```diff
@@ -225,7 +138,7 @@
                     </compilerArgs>
                 </configuration>
             </plugin>
-            <plugin>
+            <!--<plugin>
```
Comment (undx) [inline azure/azureblob/pom.xml]: remove commented code

Comment (undx) [reply azure/azureblob/pom.xml]: same remarks for plugins in pom.

Comment (undx) [reply azure/azureblob/pom.xml]: Is jacoco-maven-plugin needed while defined in parent? It's used on azurecommon & azureblob poms

File: azure/azurecommon/pom.xml [hunk 7]
```diff
@@ -13,23 +13,139 @@
...
 
-  <parent>
-    <groupId>org.talend.components</groupId>
-    <artifactId>azure</artifactId>
-    <version>1.3.0-SNAPSHOT</version>
-  </parent>
+   <parent>
+        <groupId>org.talend.components</groupId>
+        <artifactId>azure</artifactId>
+        <version>1.3.0-SNAPSHOT</version>
```
Comment (undx) [inline azure/azurecommon/pom.xml]: idem for version

File: azure/azurecommon/pom.xml [hunk 8]
```diff
@@ -13,23 +13,139 @@
...
 
-  <name>Components :: AzureCommon</name>
+   <name>Components :: AzureCommon</name>
 
-  <dependencies>
+   <dependencies>
     <dependency>
       <groupId>org.apache.httpcomponents</groupId>
       <artifactId>httpclient</artifactId>
       <version>4.5.3</version>
```
Comment (undx) [inline azure/azurecommon/pom.xml]: Maybe you should define all dependencies in azure pom (parent) in a dependencyManagement section and only mention them when required as here. All versions in parent in that case.

File: azure/azurecommon/pom.xml [hunk 9]
```diff
@@ -13,23 +13,139 @@
...
     </dependency>
   </dependencies>
+
+   <build>
+        <plugins>
+            <plugin>
+                <groupId>org.jacoco</groupId>
+                <artifactId>jacoco-maven-plugin</artifactId>
```
Comment (undx) [inline azure/azurecommon/pom.xml]: same remark for jacoco

Comment (dmytro-sylaiev) [reply azure/azurecommon/pom.xml]: In root pom jacoco is configured only for unit tests, not for integration..
That's why I needed to add my own jacoco executions

File: azure/azureblob/pom.xml (hunk 6 above)
Comment (dmytro-sylaiev) [reply azure/azureblob/pom.xml]: In root pom jacoco is configured only for unit tests, not for integration..
That's why I needed to add my own jacoco executions

File: azure/azurecommon/pom.xml (hunk 7 above)
Comment (dmytro-sylaiev) [reply azure/azurecommon/pom.xml]: We need to specify version explicitly in parent definition

**Pull request closed without merging**